### DISCUSSION TITLE
Why alerts API does not return polygon

### DISCUSSION THREAD

**Discussion by long-nguyen** [weather-gov/api#641] [Q&A] [answered]
Hi, does anyone know why some alert item does not have 'geometry' field?
API: https://api.weather.gov/alerts/active
```
"features": [
        {
            "id": "https://api.weather.gov/alerts/urn:oid:2.49.0.1.840.0.6e4c40b25fca4e54f9f785c46758df04ea96ce08.001.1",
            "type": "Feature",
            "geometry": null,
            "properties": {
                "@id": "https://api.weather.gov/alerts/urn:oid:2.49.0.1.840.0.6e4c40b25fca4e54f9f785c46758df04ea96ce08.001.1",
                "@type": "wx:Alert",
                "id": 
```

**Comment by sullynole** [comment]
There are different types of alerts.  This should help explain: https://www.weather.gov/media/documentation/docs/NWS_Geolocation.pdf

**Comment by JJKraw** [comment, marked as answer]
For alerts that do not contain an explicit polygon, you can use the geometries associated with the zone codes included in the alert. Those are found at /zones/{type}/{zoneID}

Make sure you request geo+json.

I keep a cache of these so I don't have to fetch every time.

**Reply by ephcoding** [reply]
I've noticed that the zone endpoint (zones/{type}/{id}) returns geometry as expected, but when providing multiple zones in an array  with required geometry param (zones/{type}?include_geometry=true&id=<id#1>,<id#2>) consistently returns null.

Any chance you've ran into this?  Thanks.

**Reply by JJKraw** [reply]
I have never tried that. Some of the geometries are quite large so I've never even thought about pulling more than one at a time.

**Reply by ephcoding** [reply]
really appreciate the response.  

Trying to generate a graphic similar to below. Initial thought was to use the /zones/{type} endpoint along with the <include_geometry> param to get the geometry for every county (affectedZones) in one request.

In order to recreate the watch polygon below though, it seems like the only option is to make separate requests for each affected zone and render them individually.

I really hope it's just a matter of me not finding a web service that provides geometry for each watch.

<img width="515" alt="image" src="https://github.com/weather-gov/api/assets/86921531/fe1cfc5d-781f-4c26-bfa6-f55b6882d566">

**Reply by JJKraw** [reply]
This is exactly why I cache the results. I hardly ever have to do an actual fetch.

**Reply by ephcoding** [reply]
are you just caching locally when a user initially hits your app, or along side your app?

**Reply by scadergit** [reply]
This is where the API gets zone geometry if you want to import it. https://www.weather.gov/gis/AWIPSShapefiles

**Reply by JJKraw** [reply]
I don't have an app, just a web site, so I cache on the server so the client doesn't have to wait for the server to fetch. I use Leaflet to do map rendering, so when the page loads in the browser, it includes all the javascript necessary to talk to the Leaflet API. The performance has been good.

**Reply by ephcoding** [reply]
@scadergit thanks for the link. hella weird..lol literally just used added the Public Forecast Zones to my app 5 minutes ago using the shapefiles.

Instead of calling the API for the geometry, I could probably parse all the returned UGC codes from the [affectedZones] and map those to filled svg features.

<img width="1165" alt="image" src="https://github.com/weather-gov/api/assets/86921531/02ba3624-ee3a-4210-869c-0dd2bb95339d">

**Reply by ephcoding** [reply]
@JJKraw sounds like a good way to go.  I need to get better at optimizing client-side overhead in Next.js.

I always struggled with the SDKs (Esri, OpenLayers, MapBox, Leaflet). Ended up rolling with D3.

What's your site? I'd love to check it out.  The app I'm working on is [tornadowarned.com](https://tornadowarned.com). It's OSS'd on GitHub: [modevx/tornado-warned](https://github.com/modevx/tornado-warned).

**Reply by sullynole** [reply]
Public/fire/coastal zone and county maps are also available here as services:  https://mapservices.weather.noaa.gov/static/rest/services/nws_reference_maps/nws_reference_map/MapServer

Note: WFOs generally can update their zones twice per year in spring and fall, so take that into consideration for your application.  Those changes are announced here when they occur: https://www.weather.gov/notification/#scn

**Reply by ephcoding** [reply]
@scadergit for the inoperable query params (ie. include_geometry), would it be worth me shooting an email to the [nco.ops@noaa.gov](mailto:nco.ops@noaa.gov)?

**Reply by scadergit** [reply]
Sure, but probably low priority.

**Reply by ephcoding** [reply]
@JJKraw @sullynole @scadergit thank you guys for all the help.

**Reply by JJKraw** [reply]
https://lexmaweather.info

Nothing professional. Just hobby coding for me and info for people in my town.

Example of maps / zone usage: https://lexmaweather.info/advisoryJson.php?coverage=sew
